packages/rum/src/batch.ts

```diff
@@ -0,0 +1,81 @@
+import { Batch, combine, Configuration, Context, HttpRequest } from '@datadog/browser-core'
+import { LifeCycle, LifeCycleEventType } from './lifeCycle'
+import { RumEvent, RumEventCategory } from './rum'
+
+export function startRumBatch(configuration: Configuration, lifeCycle: LifeCycle) {
+  const batch = makeRumBatch(configuration, lifeCycle)
+
+  lifeCycle.subscribe(
+    LifeCycleEventType.RUM_EVENT_COLLECTED,
+    ({ rumEvent, serverRumEvent }: { rumEvent: RumEvent; serverRumEvent: Context }) => {
+      if (rumEvent.evt.category === RumEventCategory.VIEW) {
+        batch.upsert(serverRumEvent, rumEvent.view.id)
+      } else {
+        batch.add(serverRumEvent)
+      }
+    }
```

Comment:
I though about it, IMO:
- no reason that assembly should be aware of the upsert system and the need to build a key when it is dealing with.a view. The batch itself is still not aware of rum types but the rum batch is all the extra logic around batch for rum, so it seems appropriate to me that we are dealing with that here.
- it could allow us to intercept those rumEvents for other purposes (debugging extension?)

Reply:
> no reason that assembly should be aware of the upsert system
I agree. The key could be set by the `track*` functions, sent along the `RAW_RUM_EVENT_COLLECTED` data
> it could allow us to intercept those rumEvents for other purposes (debugging extension?)
A potential debugging extension should be based on the full event containing the context. I'd be in favor of not duplicating the information if possible.

Reply:
>  The key could be set by the track* functions, sent along the RAW_RUM_EVENT_COLLECTED data

It would work but IMO: 
- it seems unnecessary to pass the key across all layers since we can compute it only when we need to use it
- the building of events (trackXXX) don't need to know how the events are stored before being sent they should not be aware of this key

Reply:
Alright then